### PR TITLE
Choose logging handler via `--log-handler` CLI option

### PR DESCRIPTION
**Pull request by cpitclaudel**
* `pelican/__init__.py` (parse_arguments): Declare new `--log-handler argument`. 
  Uses a string instead of a `type=` argument to get better error messages when passed an incorrect choice.
  (main): Pass `args.logs_handler` to `init_logging`
* pelican/log.py: Expose default log handler as `DEFAULT_LOG_HANDLER`.
  (init): Pass handler to `logging.basicConfig`.

# Pull Request Checklist

Resolves: GH-3292

- [x] Ensured **tests pass** and (if applicable) updated functional test output
- [x] Conformed to **code style guidelines** by running appropriate linting tools
- [x] Added **tests** for changed code
  **Question:** I think there's no tests for the logger?
- [x] Updated **documentation** for changed code
  **Question:** I think there's no separate documentation for the CLI, I just added an argparse `help`

### CODE REVIEW

File: pelican/__init__.py [hunk 1]
```diff
@@ -455,6 +455,17 @@ def parse_arguments(argv=None):
         ),
     )
 
+    LOGS_HANDLERS = {"plain": None, "rich": DEFAULT_LOG_HANDLER}
+    parser.add_argument(
+        "--logs-handler",
+        default="rich",
+        choices=LOGS_HANDLERS,
+        help=(
+            "Which handler to to format log messages. "
```
Comment (justinmayer) [inline pelican/__init__.py:464]: I think the word "use" is missing here? I assume this was intended to say:

> "Which handler to use to format log messages. "

Comment (cpitclaudel) [reply pelican/__init__.py]: Woops.  Thanks, fixed!

**Review by justinmayer — APPROVED** [review]
I saw that but decided that particularly discrepancy is okay for now. We shall strive for eventual consistency! 😉 

@avaris: Any thoughts on this PR?

**Comment by offbyone** [comment]
I'm curious what use this flag would be put to. It would be constructive, I think, to include documentation for how to use it, how to define a new handler, and an example custom handler.

**Comment by cpitclaudel** [comment]
> I'm curious what use this flag would be put to.

I use it in makefiles, because the forced line wrapping that the rich handler does breaks hyperlinks:

```
           ERROR    Skipping                                     contents.py:175
                    /home/xxx/xxx/xxxxxxx/xxxxx/xxxxx/xxxxxxxxxx                
                    /xxxxxxxxxxxxx/xxxxxxxx/xxxxxxxxx/xxxxxx.htm                
                    l: could not find information about 'date'           
```

With this format my terminal and my IDE don't let me click on the file in question to navigate to it.

> It would be constructive, I think, to include documentation for how to use it, how to define a new handler, and an example custom handler.

Defining a custom handler is something the Pelican devs would do, not users.  These are standard Python log handlers, so the appropriate documentation would be https://docs.python.org/3/library/logging.handlers.html

**Comment by GiovanH** [comment]
I was just about to request this. Are there any outstanding issues or is this fully approved?

**Comment by justinmayer** [comment]
Unless there are further comments from others, I intend to merge the Ruff-related PR (3239) and then this one.

**Comment by justinmayer** [comment]
Many thanks for your enhancement, Clément! 🌟

**Comment by egberts** [comment]
WOW!  Now if we can re-add the `name` back into `LOG_FORMAT` for log-handler=plain option, then the `--log-handler=plain` output would look like the following:

```console
$ python -m pelican --log-handler=plain -s /tmp/pelicanconf-invalid-PATH-1
load_source: AttributeError: 'NoneType' object has no attribute 'loader'
```
See the word `load_source` (that is in `settings.py` source file, but we now know what to look for).

Instead of the currently HEAD commit shows the following:
```console
$ python -m pelican --log-handler=plain -s /tmp/pelicanconf-invalid-PATH-1
AttributeError: 'NoneType' object has no attribute 'loader'
```

Currently, we have no idea which method/procedures (that `%(name)s` in LOG_FORMAT) provides in when an error occurs (not to mention the missing source:line-no column.  Furthermore, the patch applied has rendered the inability to use such `%(name)s` format (and possibly ignores `LOG_FORMAT`).

**Comment by cpitclaudel** [comment]
@egberts I'm sorry, I don't really understand what you're saying

**Pull request merged**